parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -139,6 +140,43 @@ public Statistics<?> build() {
     }
   }
 
+  // Builder for FLOAT16 type to handle special cases of min/max values like NaN, -0.0, and 0.0
+  private static class Float16Builder extends Builder {
+    public Float16Builder(PrimitiveType type) {
+      super(type);
+      assert type.getPrimitiveTypeName() == PrimitiveTypeName.BINARY;
+    }
+
+    @Override
+    public Statistics<?> build() {
```

Comment:
@benibus Could you please double check if this aligns with the C++ impl?

Reply:
Sorry, missed this comment somehow. Yes, this looks correct.

Reply:
Thanks for confirmation! @benibus